minio/api.py

```
@@ -302,6 +304,81 @@ def remove_bucket(self, bucket_name):
         # Make sure to purge bucket_name from region cache.
         self._delete_bucket_region(bucket_name)
 
+    def _get_bucket_policy(self, bucket_name):
+        policy_dict = {}
+        try:
+            response = self._url_open("GET",
+                                      bucket_name=bucket_name,
+                                      query={"policy": ""},
+                                      headers={})
+            policy_dict = json.loads(response.data)
+        except ResponseError as e:
+            # Ignore 'NoSuchBucketPolicy' error.
+            if e.code != 'NoSuchBucketPolicy':
+                raise
+
+        return policy_dict
+
+    def get_bucket_policy(self, bucket_name, object_prefix=""):
+        """
+        Get bucket policy of given bucket name.
+
+        :param bucket_name: Bucket name.
+        :param object_prefix: Object prefix.
+        """
+        is_valid_bucket_name(bucket_name)
+
+        policy_dict = self._get_bucket_policy(bucket_name)
+
+        # Normalize statements.
+        statements = []
+        policy.append_statements(statements, policy_dict.get('Statement', []))
+
+        return policy.get_policy(statements, bucket_name, object_prefix)
+
+    def set_bucket_policy(self, policy_access, bucket_name, object_prefix=""):
+        """
+        Set bucket policy of given bucket name and object prefix.
+
+        :param bucket_name: Bucket name.
+        :param object_prefix: Object prefix.
+        """
+        is_valid_bucket_name(bucket_name)
+
+        policy_dict = self._get_bucket_policy(bucket_name)
+        if policy_access == policy.Policy.NONE and not policy_dict:
+            return
```

We need to return an error here..We raise an exception. `NoSuchBucketPolicy`